src/test/java/org/folio/edge/sip2/repositories/FeeFinesRepositoryTests.java

```diff
@@ -0,0 +1,138 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.argThat;
+import static org.mockito.Mockito.when;
+
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Vertx;
+import io.vertx.core.impl.NoStackTraceThrowable;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import java.util.UUID;
+import org.folio.edge.sip2.session.SessionData;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith({VertxExtension.class, MockitoExtension.class})
+public class FeeFinesRepositoryTests {
+  private static final String FIELD_TOTAL_RECORDS = "totalRecords";
+  private static final String FIELD_MANUALBLOCKS = "manualblocks";
+
+  @Test
+  public void canCreateFeeFinesRepository(
+      @Mock IResourceProvider<IRequestData> mockFolioResource) {
+    final FeeFinesRepository usersRepository =
+        new FeeFinesRepository(mockFolioResource);
+
+    assertNotNull(usersRepository);
+  }
+
+  @Test
+  public void cannotCreateFeeFinesRepositoryWhenResourceProviderIsNull() {
+    final NullPointerException thrown = assertThrows(
+        NullPointerException.class,
+        () -> new FeeFinesRepository(null));
+
+    assertEquals("Resource provider cannot be null", thrown.getMessage());
+  }
+
+  @Test
+  public void canGetManualBlocksByUserId(Vertx vertx,
+      VertxTestContext testContext,
+      @Mock IResourceProvider<IRequestData> mockFolioProvider) {
+
+    final String manualBlocksResponseJson = getJsonFromFile("json/no_manual_blocks_response.json");
+    final JsonObject manualBlocksResponse = new JsonObject(manualBlocksResponseJson);
+
+    when(mockFolioProvider.retrieveResource(any()))
+        .thenReturn(Future.succeededFuture(new FolioResource(manualBlocksResponse,
+            MultiMap.caseInsensitiveMultiMap().add("x-okapi-token", "1234"))));
+
+    final SessionData sessionData = SessionData.createSession("diku", '|', false, "IBM850");
+
+    final FeeFinesRepository feeFinesRepository = new FeeFinesRepository(mockFolioProvider);
+    feeFinesRepository.getManualBlocksByUserId(UUID.randomUUID().toString(),
+        sessionData).setHandler(
+            testContext.succeeding(manualBlocks -> testContext.verify(() -> {
+              assertNotNull(manualBlocks);
+              assertNotNull(manualBlocks.getJsonArray(FIELD_MANUALBLOCKS));
+              assertEquals(0, manualBlocks.getJsonArray(FIELD_MANUALBLOCKS).size());
```

Comment:
For the intent of this test, should there be at least one manual block? 0 could also mean that no blocking at all, which might be similar to getting a NULL manualBlocks array?  

UPDATE: after writing this comment I saw how 0/NULL means in the next couple of tests. I leave this comment here to illustrate my confusion, so I'd recommend adding more descriptions for the 0/NULL in the javaDoc section of the canGetManualBlocksByUserId method.

Reply:
Yeah, I knew those names were confusing, but I kind of ignored that fact. I'll rename the tests to `canGetManualBlocksByUserIdWithNoBlocksApplied` and `canGetManualBlocksByUserIdWithBlocksApplied`. I think those names are clearer to the intent of the test.